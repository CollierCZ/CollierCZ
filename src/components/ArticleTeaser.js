import { css } from 'styled-components'
import { string } from "prop-types";
import React from "react"

const ArticleTeaser = ({ cover }) => {
  return (
    <div 
      css={css`
        width: 100%;
        height: 200px;
        background: url("${ cover }") center center no-repeat;
      `}
  />);
}

ArticleTeaser.propTypes = {
  cover: string.isRequired
}

export default ArticleTeaser;
