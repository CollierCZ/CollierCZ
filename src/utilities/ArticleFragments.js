import { graphql } from "gatsby"

export const ArticleListFragment = graphql`
  fragment ArticleListFragment on KontentItemArticle {
    fields {
      slug
      tags
      date
    }
    elements {
      metadata__description {
        value
      }
      metadata__keywords {
        value
      }
      categories {
        value {
          name
        }
      }
      title {
        value
      }
      teaser {
        value {
          url
        }
      }            
    }
  }
`