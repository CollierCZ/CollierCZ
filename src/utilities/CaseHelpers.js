const slugify = (text) => {
  return text
    .toString()
    .toLowerCase()
    .trim()
    .replace(/\s+/g, '-')       // Replace spaces with -
    .replace(/[^\w-]+/g, '')     // Remove all non-word chars
}

const titleCase = (text) => {
  return(
    text.replace(/\w\S*/g, function(txt) {
      return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
    })
  )
}

module.exports =
{
  slugify: slugify,
  titleCase: titleCase
}