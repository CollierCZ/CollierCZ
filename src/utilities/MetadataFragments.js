import { graphql } from "gatsby"

export const MetadataFragment = graphql `
  fragment MetadataFragment on KontentItemHome {
    elements {
      title {
        value
      }
      splash_image {
        value {
          url
        }
      }
      blog_logo {
        value {
          url
        }
      }
      metadata__description {
        value
      }
      socialmedia {
        value
      }
      base_url {
        value
      }
    }
  }
`

export const SocialMediaFragment = graphql `
  fragment SocialMediaFragment on KontentItemHome {
    elements {
      socialmedia {
        value
      }
    }
  }
`