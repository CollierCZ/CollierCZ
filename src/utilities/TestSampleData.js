export const author = {
  short_bio: {
    value:
      '<p>Originally from <a href="https://en.wikipedia.org/wiki/Lexington,_Massachusetts" data-new-window="true" target="_blank" rel="noopener noreferrer">Lexington</a> and now residing in <a href="https://www.gotobrno.cz/en/" data-new-window="true" target="_blank" rel="noopener noreferrer">Brno</a>, Aaron is an educator, editor, and elucidator.</p>'
  },
  picture: {
    value: [
      {
        url:
          "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/f7377b84-8ff0-4b77-bc0f-cfdc34d379cc/quixote.jpg"
      }
    ]
  },
  name: {
    value: "Aaron Collier"
  },
  url: {
    value: "aaron-collier"
  }
};

export const simpleImage = {
  description: "A cool picture",
  url: "https://example.com/"
};

export const sampleQuote = {
  system: {
    type: "quote"
  },
  elements: {
    quote: {
      resolvedData: {
        html:
          "<p>You never really understand a person until you consider things from [their] point of view â€” until you climb into [their] skin and walk around in it.</p>"
      }
    },
    source: {
      resolvedData: {
        html:
          "<p>Atticus Finch in <em>To Kill a Mockingbird </em>by Harper Lee</p>"
      }
    }
  }
};

export const sampleShowcase = {
  system: {
    type: "showcase"
  },
  elements: {
    items: {
      linked_items: [
        {
          id: "457d3938-5907-5cca-947d-bae1583e105d",
          elements: {
            link: {
              value: "https://www.youtube.com/watch?v=vj7TyUgBEW8"
            },
            name: {
              value: "Write the Docs talk"
            },
            short_description: {
              links: [
                {
                  linkId: "3cc9d486-53e2-495d-9e2b-25665e0ff29f",
                  urlSlug: "seeing-your-docs-through-different-eyes",
                  type: "article",
                  codename: "write_the_docs_2019"
                }
              ],
              resolvedData: {
                html:
                  '<p>A video of a talk I gave at the <a href="http://www.writethedocs.org/conf/prague/2019/">Write the Docs Prague 2019</a> conference. Or read a <a data-item-id="3cc9d486-53e2-495d-9e2b-25665e0ff29f" href="">text version of the talk</a>.</p>'
              }
            },
            teaser: {
              value: [
                {
                  url:
                    "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/51e9be55-8502-40ca-b374-a1305b36dc56/Slide16.PNG"
                }
              ]
            }
          },
          fields: {
            slug: "write-the-docs-talk"
          }
        },
        {
          id: "fce4fe5b-59c9-5c88-8634-16b63df49d4a",
          elements: {
            link: {
              value: "https://www.youtube.com/watch?v=MlMkAqxoPZM"
            },
            name: {
              value: "Structured content video"
            },
            short_description: {
              links: [],
              resolvedData: {
                html:
                  "<p>A video about what structured content is and how it can benefit content creators.</p>"
              }
            },
            teaser: {
              value: [
                {
                  url:
                    "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/b0481084-18a8-4cf0-ae01-d1a0dbd766ce/structured_content.jpg"
                }
              ]
            }
          },
          fields: {
            slug: "structured-content-video"
          }
        }
      ]
    }
  }
};
