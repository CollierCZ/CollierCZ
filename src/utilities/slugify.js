let slugify = (text) => {
    return text
        .toString()
        .toLowerCase()
        .trim()
        .replace(/\s+/g, '-')           // Replace spaces with -
        .replace(/[^\w-]+/g, '')       // Remove all non-word chars
}


module.exports.slugify = slugify