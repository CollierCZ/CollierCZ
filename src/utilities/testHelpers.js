import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { navigate } from "gatsby";

export const simpleRenderer = component => {
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
};
export const clickTextLink = component => {
  const mountedComponent = mount(component);
  mountedComponent
    .find("a")
    .first()
    .simulate("click");
  expect(navigate).toHaveBeenCalled();
};
