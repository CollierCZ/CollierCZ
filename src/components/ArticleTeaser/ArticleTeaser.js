import React from "react";
import { Link } from "gatsby";
import "./ArticleTeaser.css";

class ArticleTeaser extends React.Component {
  render() {
    const { cover, link } = this.props;
    let coverStyle = {backgroundImage: `url(${ cover })`};
    if (link) {
      return <Link to={link}>
          <div className="article-teaser" style={coverStyle} />
        </Link>;
    }
    return null;
  }
}

export default ArticleTeaser;
