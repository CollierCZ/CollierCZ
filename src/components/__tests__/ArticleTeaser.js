import React from "react"
import ArticleTeaser from "../ArticleTeaser"
import { simpleShallowRender } from "../../utilities/testHelpers"

const cover = "https://example.com/cool_image.png"
const link = "/articles/cool-article"

describe("ArticleTeaser", () => {
  it("renders correctly", () => {
    simpleShallowRender(<ArticleTeaser cover={cover} link={link} />)
  })
})