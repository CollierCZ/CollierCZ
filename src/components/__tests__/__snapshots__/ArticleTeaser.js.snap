// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleTeaser renders correctly 1`] = `
<div
  css={
    Array [
      "
        width: 100%;
        height: 200px;
        background: url(\\"",
      "https://example.com/cool_image.png",
      "\\") center center no-repeat;
      ",
    ]
  }
/>
`;
